src/spikeinterface/extractors/neoextractors/neobaseextractor.py

```diff
@@ -293,35 +294,125 @@ def __init__(self, block_index=None, sampling_frequency=None, use_natural_unit_i
                 use_natural_unit_ids=self.use_natural_unit_ids,
                 t_start=t_start,
                 sampling_frequency=sampling_frequency,
-                handle_spike_frame_directly=self.handle_spike_frame_directly,
+                neo_returns_timestamps=self.neo_returns_timestamps,
             )
 
             self.add_sorting_segment(sorting_segment)
 
-    def _auto_guess_sampling_frequency(self):
+    def _get_t_start_from_signal_stream(self, segment_index):
```

Comment:
@alejoe91 @samuelgarcia 
So, as per the comment  https://github.com/SpikeInterface/spikeinterface/pull/1626#discussion_r1250566361

The logic of sampling_frequency was changed.

However, getting the t_start from analog signal requires a stream_index.

Previously, this was handled by using the stream from which the sampling frequency was gotten. But now, this is not always possible. 

When this is not possible, what should we do?
@alejoe91 @samuelgarcia

Reply:
OK, after discussing with @samuelgarcia the logic would be to set if there is a single t_start that matches the frequency and otherwise set it to 0.